clang/lib/CodeGen/CoverageMappingGen.cpp

```diff
@@ -1847,30 +1916,13 @@ struct CounterCoverageMappingBuilder
     // Extract the Parent Region Counter.
     Counter ParentCnt = getRegion().getCounter();
 
-    // Extract the MCDC condition IDs (returns 0 if not needed).
```

Comment:
Since the APIs are all the same and are encapculated within MCDCBuilder, we don't need all of this here.  We can just move it down into createBranchRegion().